x-pack/plugins/observability_onboarding/public/hooks/use_install_system_integration.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback } from 'react';
+import { useTrackedPromise } from '@kbn/use-tracked-promise';
+import { i18n } from '@kbn/i18n';
+import { useKibana } from './use_kibana';
+
+// Errors
+const UNAUTHORIZED_ERROR = i18n.translate(
+  'xpack.observability_onboarding.installSystemIntegration.error.unauthorized',
+  {
+    defaultMessage:
+      'Required kibana privilege {requiredKibanaPrivileges} is missing, please add the required privilege to the role of the authenticated user.',
+    values: {
+      requiredKibanaPrivileges: "['Fleet']",
+    },
+  }
+);
+
+type ErrorType = 'AuthorizationError' | 'UnknownError';
+export interface SystemIntegrationError {
+  type: ErrorType;
+  message: string;
+}
+
+type IntegrationInstallStatus =
+  | 'installed'
+  | 'installing'
+  | 'install_failed'
+  | 'not_installed';
+
+export const useInstallSystemIntegration = ({
```

Comment:
Do we have an e2e or api test for this?

Reply:
We don't have e2e for logs onboarding yet, the reason is that UI has changed a lot during the process of development. Cypress is already configured but no tests have been included. I'm planning to address e2e tests during test plan.